### PR TITLE
Marina output descriptors

### LINKED ISSUE #320: Implement script type `Template`
related to #250

### PR DESCRIPTION
**Pull request by louisinger**
_This PR does not add any "real" features to Marina, `descriptors` is a #250 preliminary work._

TL; DR:
* bumps liquidjs-lib and ldk to latest (tag 6.0.1).
* add a `descriptors` package letting to evaluate output script descriptors templates.
* templates are tested in `tests/descriptors.spec.ts` file.

Marina descriptors and API documentation: [descriptors/doc.md](https://github.com/louisinger/marina/blob/template-parser/src/descriptors/doc.md).

it closes #320

@tiero please review

### CODE REVIEW

File: src/descriptors/doc.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+const redeemScript = result.redeemScript // OP_O .... 
+```
+
+> The `result` object value depends on the Context! 
+
+## API
+
+All descriptors can accept `xpub...` token, the public keys are replaced before parsing the template.
+
+### Raw
```
Comment (tiero) [inline src/descriptors/doc.md:35]: `raw` should only be used at top level imho, like original descriptors and...shouldn't be the scriptPubKey? (not redeem script as you mention)

Comment (tiero) [inline src/descriptors/doc.md:35]: `raw` should only be used at top level imho, like original descriptors and...shouldn't be the scriptPubKey? (not redeem script as you mention)

File: src/descriptors/doc.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+Returns:
+
+* `raw(0123456789)` redeem script as `witnesses`.
+* the segwit v0 script `(OP_0 + hash(raw(0123456789)))` as `redeemScript`.
+
+### Eltr
+
+It lets to create [P2TR](https://en.bitcoin.it/wiki/Taproot) payments. The descripor expects 2 inputs:
+
+1. The internal taproot public key. This can be replaced by an `xpub` token in order to increment it according to actual derivation path. Will fail if this is not a 64bits hex string.
```
Comment (tiero) [inline src/descriptors/doc.md:72]: we should maybe add `H_POINT` keyword in preprocessing, so we map to unspendable internal pub key for script-path-only P2TR for the three cases

Comment (tiero) [inline src/descriptors/doc.md:72]: we should maybe add `H_POINT` keyword in preprocessing, so we map to unspendable internal pub key for script-path-only P2TR for the three cases

**Comment by louisinger** [comment]
changes included in #345

**Pull request closed without merging**